src/XMakeTasks/Microsoft.CSharp.Mono.targets

```diff
@@ -0,0 +1,36 @@
+﻿<!--
+***********************************************************************************************
+Microsoft.CSharp.Mono.targets
+
+WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and have
+          created a backup copy.  Incorrect changes to this file will make it
+          impossible to load or build your projects from the command-line or the IDE.
+
+This file defines the steps in the standard build process specific for C# .NET projects.
+For example, it contains the step that actually calls the C# compiler.  The remainder
+of the build process is defined in Microsoft.Common.targets, which is imported by 
+this file.
+
+Copyright (C) Microsoft Corporation. All rights reserved.
+***********************************************************************************************
+-->
+
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+    <PropertyGroup>
+        <!-- Using _CscToolExeTmp instead of CscToolExe directly here, as we don't want to override it
+             for !mono case -->
+        <_CscToolExeTmp Condition="'$(CscToolExe)' != ''">$(CscToolExe)</_CscToolExeTmp>
+        <_CscToolExeTmp Condition="'$(CscToolExeTmp)' == '' and '$(MSBuildRuntimeType)' == 'Mono'">mcs.exe</_CscToolExeTmp>
```

Comment:
So, this file basically just sets the compiler to `mcs.exe`. I still don't really understand why this is bound to the Mono runtime. It seems like the runtime shouldn't decide the compiler. For instance, last I checked, Roslyn runs on Mono. Why can't I use Roslyn if I'm running on Mono?